cms/djangoapps/models/settings/course_metadata.py

```diff
@@ -56,21 +62,28 @@ def fetch(cls, descriptor):
         persistence and return a CourseMetadata model.
         """
         result = {}
+        metadata = cls.fetch_all(descriptor)
+        for key, value in metadata.iteritems():
+            if key in cls.filtered_list():
+                continue
+            result[key] = value
+        return result
 
+    @classmethod
+    def fetch_all(cls, descriptor):
+        """
+        Fetches all key:value pairs from persistence and returns a CourseMetadata model.
+        """
+        result = {}
```

Comment:
Yes, that is correct -- the existing fetch() was filtering out the metadata because it's fairly coupled to the view case.